### ISSUE TITLE
How to access more feed data

### ISSUE THREAD

**Issue by shairyar** [alexdebril/FeedIoBundle#20]
Hi,

At the moment the result of a fetch gives `title`, `link`, `description` and so on, what if i want to see what else is returned in rss how can i see that information and then add it to the database?

Baig

**Comment by alexdebril** [comment]
Hi,

You can do this using [Node](https://github.com/alexdebril/feed-io/blob/master/src/FeedIo/Feed/Node.php)'s methods, especially : 

- listElements() : returns every elements' name
- getElementIterator() : returns elements filtered by name
- getAllElements() : returns all elements

For instance : 

```php

$url = "http://php.net/feed.atom";
$feedIo = $this->container->get('feedio');
$result = $feedIo->read($url);

foreach( $result->getFeed() as $item ) {
  $elementNames = $item->listElements();
  foreach( $elementNames as $elementName ) {
      $elements = $item->getElementIterator($elementName);
      foreach ($elements as $element) {
          var_dump($element->getName());
          var_dump($element->getValue());
      }
  }
}
```

FeedIoBundle is built on top of [feed-io](https://github.com/alexdebril/feed-io), you may want to browse its repository to get more information on how it works.

I hope that helped !

Alex

**Comment by shairyar** [comment]
Hi,

Many thanks, this does help alot.

Following your code only gives me the first element for the RSS, why does it not move on to the next element?

For example this is what I see as an output, you will notice that it only gives `published`
```

object(FeedIo\Feed\Node\Element)[540]
  protected 'name' => string 'published' (length=9)
  protected 'value' => string '2016-04-18T10:35:47+02:00' (length=25)
  protected 'attributes' => 
    array (size=0)
      empty
object(FeedIo\Feed\Node\Element)[548]
  protected 'name' => string 'published' (length=9)
  protected 'value' => string '2016-04-05T11:30:00+00:00' (length=25)
  protected 'attributes' => 
    array (size=0)
      empty
object(FeedIo\Feed\Node\Element)[558]
  protected 'name' => string 'published' (length=9)
  protected 'value' => string '2016-03-31T16:28:02-07:00' (length=25)
  protected 'attributes' => 
    array (size=0)
      empty
```


Baig

**Comment by alexdebril** [comment]
Hi,

I think you found a bug in listElements() !

The result seems better with getAllElements() : 

```php
$url = "http://php.net/feed.atom";
$feedIo = $this->container->get('feedio');
$result = $feedIo->read($url);

foreach( $result->getFeed() as $item ) {
  var_dump($item->getAllElements());
}
```

Alex

**Comment by shairyar** [comment]
Hi,

Many thanks,

Yes the `getAllElements()` worked and I can now see all elements, inside the controller it seems to be working fine, now how do I setup things so each time i add or update new URL it gets the elements that i need rather than the default that has been setup.

Baig

**Comment by shairyar** [comment]
Please ignore my above questions, I now understand that all extra elements i needed are actually being saved inside table `element` which is really amazing. Up till now I have been trying to do what the bundle already does 👍 

However I am struggling a bit on how to `var_dump` all items with their elements.

**Comment by shairyar** [comment]
Hi @alexdebril,

Any tips on this for me? Because of `element_set_id ` being missing on `element` table, when i query the item it does not bring the related elements, if i manually add the `element_set_id` then it works fine.

Will appreciate any info on it. Am i missing something or this is a bug?

Baig

**Comment by alexdebril** [comment]
Hi,

I think this is a bug. It probably comes from the mapping between [ElementSet](https://github.com/alexdebril/FeedIoBundle/blob/master/Entity/ElementSet.php) and [Element](https://github.com/alexdebril/FeedIoBundle/blob/master/Entity/Element.php).

It may take some time to solve this, I'll let you know when I'll found what's going on between those two tables.

Alex

**Comment by shairyar** [comment]
Many thanks, I will really appreciate a earliest response :)

**Comment by shairyar** [comment]
by any chance is there any update on this

**Comment by alexdebril** [comment]
No, I'm not available to dig into this issue for the moment, I may have some time to do so next week. Meanwhile, you can read [Doctrine's documentation](http://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/reference/association-mapping.html) if you are willing to fix the bug and submit a pull request.

**Comment by shairyar** [comment]
The relationship seems alright to me i looked into the relationship between `item` and `media` as they also share `OneToMany` and `ManyToOne` relationship like `element` and `elementSet` but look same to me :)

**Comment by shairyar** [comment]
Hi,

I ended up fixing the issue inside `ElementSet` you have

```
    public function addElement(ElementInterface $element)
    {
        $this->elements[] = $element;
        return $this;
    }
```
I changed this to as following and it worked.

```
    public function addElement(ElementInterface $element)
    {
        $this->elements[] = $element;
        $element->setElementSet($this);

        return $this;
    }
```

**Comment by alexdebril** [comment]
Great, thank you very much !

I'll set a new release with this fix.

**Comment by alexdebril** [comment]
fixed by v3.1.5
